### PR TITLE
Match X.com URLs

### PR DESCRIPTION
**Pull request by pratham-tharwani**

### CODE REVIEW

**Comment by pratham-tharwani** [comment]
Honestly the way I did this is bad and I hate it but I don't wanna refactor your code

File: main.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import config
 
 twitter_url_regex = re.compile(
-    r"(?<!<)https?://(?:mobile\.)?twitter\.com/([^/]+)/status/(\d+)(?!\S*>)", re.I
+    r"(?<!<)https?://(?:mobile\.)?(twitter|x)\.com/([^/]+)/status/(\d+)(?!\S*>)", re.I
```
Comment (petzku) [inline main.py]: This should be a non-capturing group, so that the iteration on L29 unpacks the tuples correctly.

File: main.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def should_spoiler(message):
 
 def should_nag(message):
     if not (
-        "//twitter.com" in message.content or "mobile.twitter.com" in message.content
+        re.search("(\/\/|mobile\.)(twitter|x)\.com", message.content)
```
Comment (petzku) [inline main.py]: I'm not sure if this is how I would have done this, but it works well enough. `\/\/` is unnecessary and could just be `//`.

The line should be wrapped onto the `if not` line, according to [the Black formatter](https://github.com/psf/black) (which is not documented anywhere, aside from a commit saying "format files with Black"... I should probably fix that).

Also, while this _works_, it's good practice (and convention) to use raw strings (`r"..."`) for regex, as backslashes are also used for normal string escapes in Python. Here, `\.` is not a recognized escape, so it gets passed along literally, but it's a good idea to be consistent about it.

File: main.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def should_nag(message):
 
 def _is_video_tweet(embed):
     print(embed)
-    return embed.video and "twitter.com" in embed.url
+    return embed.video and (True if re.search("(x|twitter)\.com", embed.url) else False)
```
Comment (petzku) [inline main.py]: Similar to L65, should use raw strings for regex by convention. Also, for what it's worth, two simple string searches (as in, ``"twitter.com" in embed.url or "x.com" in embed.url``) is still about 5-10x faster than a regex search (not that this is terribly performance-critical either way).

**Review by petzku — CHANGES_REQUESTED** [review]
Thank you for the PR! I can handle the comments here tomorrow, if you don't want to do it sooner.

**Comment by pratham-tharwani** [comment]
Alright I made a couple of the changes you requested. Have a look whenever you're free, I won't be online for a while now.

**Review by petzku — APPROVED** [review]
Looks good!

**Pull request merged**